embedded/sql/catalog.go

```diff
@@ -251,10 +253,19 @@ func (catlg *Catalog) newTable(name string, colsSpec []*ColSpec) (table *Table,
 		return nil, fmt.Errorf("%w (%s)", ErrTableAlreadyExists, name)
 	}
 
-	id := len(catlg.tables) + 1
+	// Generate a new ID for the table by incrementing the 'tableCount' variable of the 'catalog' instance.
+	id := (catlg.tableCount + 1)
+
+	// This code is attempting to check if a table with the given id already exists in the Catalog.
+	// If the function returns nil for err, it means that the table already exists and the function
+	// should return an error indicating that the table cannot be created again.
+	_, err = catlg.GetTableByID(id)
```

Comment:
this should never happen actually, if it does it'll be a bug, right?

Reply:
It was a bug in the previous way the tables were being generated, hence I'd added it. At least it helps catch it